packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,12 +1,119 @@
-import { captureException } from '@sentry/core';
+import { captureException, getCurrentHub, startTransaction } from '@sentry/core';
 import { getActiveTransaction } from '@sentry/tracing';
+import { Transaction } from '@sentry/types';
+import { fill } from '@sentry/utils';
+import * as domain from 'domain';
+import { IncomingMessage, ServerResponse } from 'http';
+
+declare module 'http' {
+  interface IncomingMessage {
+    _sentryTransaction?: Transaction;
+  }
+}
+
+function getTransactionFromRequest(req: IncomingMessage): Transaction | undefined {
+  return req._sentryTransaction;
+}
+
+function setTransactionOnRequest(transaction: Transaction, req: IncomingMessage): void {
+  req._sentryTransaction = transaction;
+}
+
+function autoEndTransactionOnResponseEnd(transaction: Transaction, res: ServerResponse): void {
+  fill(res, 'end', (originalEnd: ServerResponse['end']) => {
+    return function (this: unknown, ...endArguments: Parameters<ServerResponse['end']>) {
+      transaction.finish();
+      return originalEnd.call(this, ...endArguments);
+    };
+  });
+}
+
+/**
+ * Wraps a function that potentially throws. If it does, the error is passed to `captureException` and retrhrown.
+ */
+export function withErrorInstrumentation<F extends (...args: any[]) => any>(
+  origFunction: F,
+): (...params: Parameters<F>) => ReturnType<F> {
```

Comment:
IDK if this would work, but could this just be `F`?

Reply:
Been there done that. Probably? But I don't have a clue how to fix the TS error that appears if I replace it with F.

```
Type '(this: unknown, ...origFunctionArguments: Parameters<F>) => ReturnType<F>' is not assignable to type 'F'.
  '(this: unknown, ...origFunctionArguments: Parameters<F>) => ReturnType<F>' is assignable to the constraint of type 'F', but 'F' could be instantiated with a different subtype of constraint '(...args: any[]) => any'.
```

whatever that means

Reply:
Huh. Interesting. I didn’t know what that meant, either, so I googled it, and found [a very thorough SO answer](https://stackoverflow.com/a/59363875/), which definitely helped me understand the error, but not why it applies in this situation. 

Every time I think I understand TS...